### ISSUE TITLE
=dev-libs/efl-9999[egl,gles,opengl]: error: Full OpenGL with EGL is not supported

### ISSUE THREAD

**Issue by waebbl** [rafspiny/enlightenment-live#62]
When trying to compile efl with USE="egl gles opengl" it fails during configure phase with
```
configure: Start Evas checks
checking whether Cocoa framework is supported... no
configure: error: Full OpenGL with EGL is not supported, please add --with-opengl=es to your configure options to switch to EGL + OpenGL ES.

!!! Please attach the following file when seeking support:
!!! /var/tmp/portage/dev-libs/efl-9999/work/efl-9999/config.log
 * ERROR: dev-libs/efl-9999::enlightenment-live failed (configure phase):
 *   econf failed
 * 
 * Call stack:
 *               ebuild.sh, line  124:  Called src_configure
 *             environment, line 3492:  Called econf '--enable-image-loader-generic' '--enable-image-loader-jpeg' '--enable-image-loader-png' '--enable-image-loader-bmp' '--enable-image-loader-wbmp' '--enable-image-loader-eet' '--enable-image-loader-gif' '--enable-image-loader-ico' '--enable-image-loader-jp2k' '--enable-librsvg' '--enable-image-loader-tga' '--enable-cserve' '--enable-elput' '--enable-multisense' '--enable-libmount' '--enable-libeeze' '--enable-threads' '--enable-xinput22' '--enable-liblz4' '--disable-doc' '--disable-gesture' '--disable-gstreamer' '--disable-tizen' '--disable-wayland-ivi-shell' '--enable-install-eo-files' '--disable-doc' '--disable-lua-old' '--disable-pixman' '--disable-pixman-font' '--disable-pixman-rect' '--disable-pixman-line' '--disable-pixman-poly' '--disable-pixman-image' '--disable-pixman-image-scale-sample' '--enable-image-loader-pmaps' '--enable-spectre' '--enable-image-loader-psd' '--enable-pulseaudio' '--enable-libraw' '--disable-scim' '--disable-sdl' '--enable-audio' '--enable-systemd' '--enable-image-loader-tiff' '--disable-tslib' '--disable-avahi' '--enable-image-loader-dds' '--enable-drm' '--enable-elput' '--enable-egl' '--enable-fb' '--enable-fontconfig' '--disable-fribidi' '--enable-gstreamer1' '--enable-harfbuzz' '--enable-hyphen' '--disable-ibus' '--enable-libuv' '--enable-lua-old' '--disable-neon' '--enable-nls' '--enable-poppler' '--enable-physics' '--enable-spectre' '--enable-image-loader-pmaps' '--enable-image-loader-psd' '--enable-pulseaudio' '--disable-scim' '--disable-sdl' '--enable-librsvg' '--enable-systemd' '--enable-image-loader-tga' '--enable-image-loader-tiff' '--disable-tslib' '--enable-v4l2' '--disable-valgrind' '--enable-vnc-server' '--disable-wayland' '--enable-image-loader-webp' '--enable-xcf' '--disable-xim' '--disable-xine' '--enable-image-loader-xpm' '--with-crypto=openssl' '--with-glib=no' '--with-js=none' '--with-net-control=connman' '--with-opengl=full' '--with-profile=release' '--with-x11=xlib' '--with-x' '--enable-i-really-know-what-i-am-doing-and-that-this-will-probably-break-things-and-i-will-fix-them-myself-and-send-patches-abb' '--enable-eglfs' '--enable-xpresent' '--with-opengl=full' '--enable-libvlc'
 *        phase-helpers.sh, line  718:  Called __helpers_die 'econf failed'
 *   isolated-functions.sh, line  119:  Called die
 * The specific snippet of code:
 *              die "$@"
```

It might be able to solve this by adding `opengl? ( !egl !gles)` to `REQUIRED_USE`, or trying to pass the option as described in the output.
By removing `egl` and `gles` from USE, efl compiled successfully.

**Comment by waebbl** [comment]
Note, that as of today, upstream has changed their branch to use meson instead of autotools, the ebuild is no longer working.

**Comment by rafspiny** [comment]
@waebbl I will take a look at the first problem. 
About meson, I am gonna fix it this week. Related #63

**Comment by waebbl** [comment]
@rafspiny maybe the first issue is no longer valid by moving to meson. At a first glance, I noted at least having opengl and gles support at the same time, should get retracted by the meson file.

**Comment by rafspiny** [comment]
@waebbl I actually think that the `gles` USE flag can be safely removed. 
From the meson_options.txt file you can only specify the following options:
 - **full**
 - **es-egl**

I am thinking of passing **full** if the `opengl` flag is specified and **es-egl** if the `egl` is.

Alternatively, I can add a check to make sure both flag must be enabled to use `es-egl`. This may be a more suitable solution.

**Comment by rafspiny** [comment]
@waebbl This is the relevant section of the meson file: https://github.com/Enlightenment/efl/blob/master/src/lib/evas/meson.build#L160

**Comment by waebbl** [comment]
@rafspiny I agree with you, from what it looks like in the meson configuration files. `Gles` seems to be only viable in combination if `egl`. As the `gles` USE flag seems to be used in a couple of packages, I would go for the latter way, thus, not removing the `gles` USE flag, but adding a `REQUIRED_USE` flag to check, both are set.
I'm sorry, I won't find the time to test your changes before sunday.

**Comment by rafspiny** [comment]
@waebbl Thank you for helping me to improve the ebuilds.

At the moment I am allowing any combination to be specified. 

1. USE="opengl -egl -gles" => opengl=full
2. USE="opengl **egl** -gles" => opengl=full
3. USE="opengl -egl **gles**" => opengl=full
4. USE="-opengl egl gles" => opengl=es-egl

The reasons why I rather prefer to allow combination number 2 and 3 and force them to `opengl=full` instead of forbid these two combinations are:

-  that I do not have a good understanding of how these use flags are used in other packages. maybe there are systems out there where they only have egl flag. I simply do not know.
- It looks like, on modern systems, full gl probably includes egl/g

Unless there is good reason to forbid combination 2 and 3, I think I will stick to this approach.
This should be closed with #66

**Comment by rafspiny** [comment]
@waebbl I am closing this one. Feel free to reopen it if needed.
